### PR TITLE
Simplify `Middleware`

### PR DESCRIPTION
**Pull request by wojpawlik**
# Description

Dropped `Middleware` namespace and `Middleware.Ext*`, and dealt with most of the consequences. Please take over @KnorpelSenf.

## Type of change

- Breaking change (fix or feature that would cause existing functionality to not work as expected)

# How Has This Been Tested?

# Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

**Comment by KnorpelSenf** [comment]
@wojpawlik please review (I cannot request you because you authored this)

File: src/scenes/wizard/index.ts [hunk 1]
```diff
@@ -33,10 +33,11 @@ export class WizardScene<C extends WizardContext = WizardContext>
   }
 
   middleware() {
-    return Composer.compose<C, WizardContext>([
+    return Composer.compose<C>([
       (ctx, next) => {
-        const wizard = new WizardContextWizard<C>(ctx, this.steps)
-        return next(Object.assign(ctx, { wizard }))
+        const steps = this.steps as Array<MiddlewareFn<WizardContext>>
+        ctx.wizard = new WizardContextWizard<WizardContext>(ctx, steps)
```
Comment (wojpawlik) [inline src/scenes/wizard/index.ts:39]: Why is the cast necessary?

Comment (KnorpelSenf) [reply src/scenes/wizard/index.ts:39]: For the same reason as here:
https://github.com/telegraf/telegraf/pull/1251/files#diff-e6b547a30db4384fb39b5ef00d301d72dc76758ce50a4a5c1c216820c73cb863R35-R38

Just that we cannot cast there, while this is possible here.

Comment (wojpawlik) [reply src/scenes/wizard/index.ts:39]: Alright. Comment would be nice. LGTM other than that.

**Review by KnorpelSenf — APPROVED** [review]

**Review by KnorpelSenf — APPROVED** [review]

**Pull request merged**